pinot-core/src/test/java/com/linkedin/pinot/core/io/readerwriter/impl/ReaderWriterThreadSafetyTests.java

```diff
@@ -0,0 +1,288 @@
+
+  private volatile int _numEntries = 0;
+
+  @BeforeClass
+  public void setup() {
+    // Printing to stdout so that we can reproduce a test with the seed if it fails.
+    System.out.println("Random seed = " + RANDOM_SEED);
+  }
+
+  @Test
+  public void testFixedByteSingleColumnSingleValueReaderWriter() throws Exception {
+    testFixedByteSingleColumnSingleValueReaderWriter(FieldSpec.DataType.INT);
+    testFixedByteSingleColumnSingleValueReaderWriter(FieldSpec.DataType.LONG);
+    testFixedByteSingleColumnSingleValueReaderWriter(FieldSpec.DataType.FLOAT);
+    testFixedByteSingleColumnSingleValueReaderWriter(FieldSpec.DataType.DOUBLE);
+  }
+
+  private void testFixedByteSingleColumnSingleValueReaderWriter(FieldSpec.DataType dataType) throws Exception {
+
+    int initialChunkSize = RANDOM.nextInt(5000) + 5000;  // some number between 5k and 10k
+    for (int chunkSize = initialChunkSize; chunkSize < MAX_NUM_ENTRIES; chunkSize *= 2) {
+      final PinotDataBufferMemoryManager memoryManager = new DirectMemoryManager(ReaderWriterThreadSafetyTests.class.getName());
+      _numEntries = 0;
+      FixedByteSingleColumnSingleValueReaderWriter readerWriter =
+          new FixedByteSingleColumnSingleValueReaderWriter(chunkSize, getColumnSize(dataType), memoryManager, "test");
+      SVWriter writer = new SVWriter(readerWriter, dataType);
+      Future<Void>[] readerFutures = new Future[NUM_READERS];
+      for (int i = 0; i < NUM_READERS; i++) {
+        SVReader reader = new SVReader(readerWriter, dataType);
+        readerFutures[i] = _executorService.submit(reader);
+      }
+      Future<Void> writerFuture = _executorService.submit(writer);
+
+      writerFuture.get();
+      for (int i = 0; i < NUM_READERS; i++) {
+        readerFutures[i].get();
+      }
+      readerWriter.close();
+      memoryManager.close();
+    }
+  }
+
+  private int getColumnSize(FieldSpec.DataType dataType) throws Exception {
+    switch (dataType) {
+      case INT:
+        return V1Constants.Numbers.INTEGER_SIZE;
+      case LONG:
+        return V1Constants.Numbers.LONG_SIZE;
+      case DOUBLE:
+        return V1Constants.Numbers.DOUBLE_SIZE;
+      case FLOAT:
+        return V1Constants.Numbers.FLOAT_SIZE;
+      default:
+        throw new IllegalArgumentException("Unsupported type " + dataType.name());
+    }
+  }
+
+  class SVWriter implements Callable<Void>{
+    private final FixedByteSingleColumnSingleValueReaderWriter _readerWriter;
+    private final FieldSpec.DataType _dataType;
+    SVWriter(FixedByteSingleColumnSingleValueReaderWriter readerWriter, FieldSpec.DataType dataType) {
+      _readerWriter = readerWriter;
+      _dataType = dataType;
+    }
+
+    @Override
+    public Void call() throws Exception {
+      for (int entry = 0; entry < MAX_NUM_ENTRIES; entry++) {
+        writeEntry(entry);
+      }
+      return null;
+    }
+
+    private void writeEntry(int entry) {
+      switch (_dataType) {
```

Comment:
we have such switch statements at many places in the code. How do we know that we have handled all possible types. Also, If a new type is added, how do we know all the data types are handled properly.